src/server.rs

```diff
@@ -204,13 +204,19 @@ struct LspServerOptions {
 #[allow(dead_code)]
 pub struct LspServer {
     store: FileStore,
+    // TODO(nathanielc): This should not be an option,
+    // rather we should always expect a client.
+    // Current testing code doesn't know how to mock a client
+    // so we make it optional for now.
```

Comment:
Yeah, I ran into this recently as well. I'm not entirely sure how to make this work, and is actually a big driving force behind why the `LspServer` doesn't currently keep a `client` around. I'll dig in and see if I can't find a solution.